src/Analyzers/CSharp/Analyzers/SimplifyLinqExpressions/CSharpSimplifyLinqExpressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var whereMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+            var linqMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+            var validLinqCalls = new List<string> { "First", "Last", "Single", "Any", "Count", "SingleOrDefault", "FirstOrDefault", "LastOrDefault" };
+
+            var enumNamedType = context.Compilation.GetTypeByMetadataName("System.Linq.Enumerable");
+            var queryNamedType = context.Compilation.GetTypeByMetadataName("System.Linq.Queryable");
+            var enumMethods = enumNamedType?.GetMembers("Where").OfType<IMethodSymbol>();
+            var queryMethods = queryNamedType?.GetMembers("Where").OfType<IMethodSymbol>();
+            AddIfNotNull(whereMethods, enumMethods);
+            AddIfNotNull(whereMethods, queryMethods);
+
+            // add all valid linq calls
+            foreach (var id in validLinqCalls)
+            {
+                enumMethods = enumNamedType?.GetMembers(id).OfType<IMethodSymbol>();
+                queryMethods = queryNamedType?.GetMembers(id).OfType<IMethodSymbol>();
+                AddIfNotNull(linqMethods, enumMethods);
+                AddIfNotNull(linqMethods, queryMethods);
+            }
+
+            if (whereMethods.Count > 0 && linqMethods.Count > 0)
+            {
+                context.RegisterOperationAction(
+                    context => AnalyzeAction(
+                        context, whereMethods.ToImmutable(), linqMethods.ToImmutable()), OperationKind.Invocation);
+            }
+
+            static void AddIfNotNull(ImmutableHashSet<IMethodSymbol>.Builder set, IEnumerable<IMethodSymbol>? others)
+            {
+                if (others != null)
+                {
+                    set.UnionWith(others);
+                }
+            }
+        }
+
+        public void AnalyzeAction(OperationAnalysisContext context, ImmutableHashSet<IMethodSymbol> whereMethods, ImmutableHashSet<IMethodSymbol> linqMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var semanticModel = context.Operation.SemanticModel;
+            // Check that .Where(...) is not user defined
+            var child = invocationOperation.Children.FirstOrDefault(c => c is IArgumentOperation);
+            var whereClause = child?.Children.FirstOrDefault(c => c is IInvocationOperation);
+            var isLambda = whereClause?.Children.Any(c => c.Syntax.IsKind(SyntaxKind.Argument) &&
```

Comment:
instead of `?.` just null check in the line before and bail (applies to other lines above too).